### PR TITLE
Add examples for all slice functions

### PR DESCRIPTION
**Pull request by jcharum**
Examples are nice documentation. Make an example for each slice function in the hopes of making their behavior more clear, especially to first-time users.

### CODE REVIEW

File: slice_test.go [hunk 1]
```diff
@@ -919,3 +923,394 @@ func TestMetrics(t *testing.T) {
...
+			"sed",
+			"do",
+			"eiusmod",
+		},
+	)
+	type accum struct {
+		ready bool
+		// sum is the sum of integers in the second column.
+		sum int
+		// product is the prouct of integers in the second column.
```
Comment (josh-newman) [inline slice_test.go:1012]: typo ("prouct")

File: reshuffle_test.go [hunk 2]
```diff
@@ -113,3 +114,196 @@ func TestRepartitionType(t *testing.T) {
...
+	// 5
+	// 6
+	// ## row count per shard
+	// shard:0 count:2
+	// shard:1 count:1
+	// shard:2 count:1
+	// shard:3 count:2
+}
+
+func ExampleReshuffle() {
```
Comment (josh-newman) [inline reshuffle_test.go:248]: Some of these examples are pretty detailed in setup. I think that's fine; readers will come here after they're already invested. But to make it a little easier, how about a quick (1 sentence) summary of the plan in a comment at the top, just so they have a preview of that they'll be looking for? For the ones that are longer than ~1 screen.

"ExampleReshuffle counts rows per shard before and after a reshuffle, showing same-keyed rows all go to the same shard."

File: slice_test.go [hunk 3]
```diff
@@ -919,3 +923,394 @@ func TestMetrics(t *testing.T) {
...
+	// 0 0 zero.squared
+	// 1 1 one.squared
+	// 2 4 two.squared
+	// 3 9 three.squared
+}
+
+func ExamplePrefixed() {
+	slice := bigslice.Const(2,
+		[]string{"a", "a", "a", "a", "a", "b", "b", "b", "b", "c", "c"},
+		[]string{"c", "a", "b", "c", "c", "b", "a", "a", "a", "a", "c"},
```
Comment (josh-newman) [inline slice_test.go:1069]: Optional: Use different letters here, or maybe two categories of objects (cat, dog, etc. vs. red, green, blue) for the two slice columns so the output is a little less alphabet-soupy.

File: reshuffle_test.go (hunk 2 above)
Comment (jcharum) [reply reshuffle_test.go:255]: I gave it a whirl.

File: slice_test.go (hunk 3 above)
Comment (jcharum) [reply slice_test.go]: I switched it to count the number of pet (type, name) pairs.

**Pull request merged**